sefaria/model/text.py

```diff
@@ -2782,6 +2789,48 @@ def subrefs(self, length):
             l.append(self.subref(i + 1))
         return l
 
+    def all_segment_refs(self):
+        seg_refs = []
+        if self.is_book_level():
+            return self.index.all_segment_refs()
+
+        sections = self.all_section_refs() if self.is_section_level() == False else [self]
+
+        for sec_ref in sections:
+            seg_refs += sec_ref.all_subrefs()
```

Comment:
This will return all *possible* segment refs, not all *populated* segment refs.  Is that the intention?  In any case, it should be documented.